packages/core/test/unit/configuration.test.ts

```diff
@@ -326,7 +326,8 @@ describe('Sequelize constructor', () => {
       const sequelize = new Sequelize('sqlite:/home/abs/dbname.db', { storage: '/completely/different/path.db' });
       const options = sequelize.options;
       expect(options.dialect).to.equal('sqlite');
-      expect(options.storage).to.equal(path.resolve('/completely/different/path.db'));
+      // this fix is likely not valid, and should be updated to a more proper solution
+      expect(options.storage).to.equal('/completely/different/path.db');
```

Comment:
The `path.resolve` was added in https://github.com/sequelize/sequelize/pull/10047 with no real explanation why. For me it's strange that it the other resolves do work and this one does not

Reply:
I assume it's an unindented side-affect of the `{ storage: '/completely/different/path.db' }` argument since all of the other test cases do not have any `{ storage: string }` arguments.
Which to me would imply that `path.resolve` isn't being ran on the storage parameter like it probably should

Reply:
Also on unix, resolving that path doesn't change it, but on windows it does. So this error would go undetected
![image](https://github.com/sequelize/sequelize/assets/11359344/12b5d79b-2ce6-457d-85a2-3b2bff795d48)

Reply:
Also on unix, resolving that path doesn't change it, but on windows it does. So this error would go undetected
![image](https://github.com/sequelize/sequelize/assets/11359344/5603f3d6-8314-462f-a75a-de4bd4034bb1)

Reply:
Also on unix, resolving that path doesn't change it, but on windows it does. So this error would go undetected
![image](https://github.com/sequelize/sequelize/assets/11359344/bd0b35e8-32ea-4b57-a0c4-14dd1a30c3e8)

Reply:
So realistically I should probably add the resolve back in, since it should be correctly triggering a fail. But it's also not an issue of this commit, and I don't feel comfortable attempting to fix a bug in the `Sequelize` class when I have little knowledge on this project.

But that then means I can't have the test properly configured, and merge this without a failing test... Unless you or someone else wants to perform edits in my branch, or pause this PR until this bug is fixed else where and can be merged in.

Reply:
Let's add a TODO here that that test should be using `path.resolve` instead and then take a look at it in a future PR. This should not block the merger of this PR imo

Reply:
Added a comment linking to [here](https://github.com/sequelize/sequelize/pull/16283#discussion_r1268796636) for context in that file in case the reasoning is lost to time if it gets delayed